### PR TITLE
Karl's getting more mature and can handle Claires.

### PR DESCRIPTION
**Pull request by dkorolev**
karl

### CODE REVIEW

**Comment by dkorolev** [comment]
Yay!

karl-dependencies

**Comment by dkorolev** [comment]
Minimalistic:

karl-minimalistic

Unresolved:

karl-unresolved-ok

**Comment by dkorolev** [comment]
@mzhurovich This is unit-tested, regression-tested (via `examples/Karl`) and battle-tested (RS).

PTAL!

File: scripts/gen-current-build.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@ if [[ $? -ne 0 ]]; then
 	exit 1
 fi
 
-COMPILER_INFO="$($CPLUSPLUS -v 2>&1)"
+COMPILER_INFO="$($CPLUSPLUS --version 2>&1)"
```
Comment (mzhurovich) [inline scripts/gen-current-build.sh]: @dkorolev Btw, `-v` for GCC results in more detailed information:
```
$ g++ --version
g++ (Ubuntu 4.8.2-19ubuntu1) 4.8.2
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/4.8/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 4.8.2-19ubuntu1' --with-bugurl=file:///usr/share/doc/gcc-4.8/README.Bugs --enable-languages=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.8 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.8 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --disable-libmudflap --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.8-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.8-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.8-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.8.2 (Ubuntu 4.8.2-19ubuntu1)
```

Comment (dkorolev) [inline scripts/gen-current-build.sh]: Good point. We can move back to `-v` upon cleaning up `build_info` from the stream/log of Karl.

**Pull request merged**